Legion/VaultOfTheWardens/Tormentorum.lua

```diff
@@ -115,3 +123,63 @@ end
 function mod:FrighteningShout(args)
 	self:Message(args.spellId, "Urgent", "Warning", CL.casting:format(args.spellName))
 end
+
+do
+	local prev = 0
+	function mod:ShadowCrashDamage(args)
+		local t = GetTime()
+		if self:Me(args.destGUID) and t-prev > 1.5 then
+			prev = t
+			self:Message(args.spellId, "Personal", "Alert", CL.underyou:format(args.spellName))
+		end
+	end
+end
+
+do
+	local prev = 0
+
+	local function printStacks(self, key)
+		local meOnly = self:CheckOption(key, "ME_ONLY")
+		if meOnly then
+			local stacks = select(4, UnitDebuff("player", self:SpellName(key))) or 0
+			if stacks > 6 then
+				self:StackMessage(key, UnitName("player"), stacks, "Urgent", stacks < 9 and "Alarm" or "Warning")
+			end
+		else
+			local highestStacks, stacksOnMe = 0, 0
+			local affectedPlayers = {}
+			for unit in self:IterateGroup() do
+				local stacks = select(4, UnitDebuff(unit, self:SpellName(key))) or 0
+				if stacks > 6 then
+					affectedPlayers[#affectedPlayers + 1] = { name = UnitName(unit), stacks = stacks }
+					if highestStacks < stacks then
+						highestStacks = stacks
+					end
+					if UnitIsUnit("player", unit) then
+						stacksOnMe = stacks
+					end
+				end
+			end
+			if #affectedPlayers == 1 then
+				self:StackMessage(key, affectedPlayers[1].name, affectedPlayers[1].stacks, "Urgent", (UnitName("player") == affectedPlayers[1].name or self:Dispeller("magic")) and (affectedPlayers[1].stacks < 9 and "Alarm" or "Warning"))
+			else
+				for i, v in ipairs(list) do
+					affectedPlayers[i] = L.count:format(self:ColorName(v.name), v.stacks)
+				end
+				local list = table.concat(affectedPlayers, ", ")
+				local stacksOfInterest = self:Dispeller("magic") and highestStacks or stacksOnMe
+				self:Message(key, "Urgent", stacksOfInterest < 9 and "Alarm" or stacksOfInterest > 6 and "Warning", L.other:format(self:SpellName(key), list))
+			end
+		end
+	end
+
+	function mod:FleshToStone(args)
+		local t = GetTime()   -- 2 of these adds spawn, they will finish their casts
+		if t - prev < 5 then  -- roughly at the same time, throttling to avoid being spammy
+			return
+		end
+		prev = t

```

Comment:
This throttle doesn't make sense now. Let's say the first wave of debuffs gets applied at 0 and the second wave at 0.5. You'd only print the debuffs at 0.3. And never the second wave.
To solve this, you should schedule the timer below long enough to catch both casts (0.3s might be enough, check logs), so use a throttle like this:

    if not scheduled then
    	scheduled = self:ScheduleTimer(...`

Reply:
Do I need to set `scheduled` to `false` or `nil` at the end of the scheduled function?

Reply:
Yes